employee-rostering-webapp/src/test/java/org/optaweb/employeerostering/webapp/shift/ShiftRestServiceIT.java

```diff
@@ -0,0 +1,189 @@
+
+        List<ShiftView> shifts = shiftRestService.getShifts(TENANT_ID);
+        assertClientResponseOk();
+        assertThat(shifts)
+                .usingComparatorForElementFieldsWithType((a, b) -> a.compareTo(b), Integer.class)
+                .usingComparatorForElementFieldsWithType((a, b) -> a.compareTo(b), Long.class)
+                .usingComparatorForElementFieldsWithType((a, b) -> a.compareTo(b), LocalDateTime.class)
+                .usingElementComparatorOnFields(shiftViewNonIndictmentFields);
+
+        ShiftView testUpdateShift = shifts.get(0);
+        testUpdateShift.setEmployeeId(employee.getId());
+        testUpdateShift.setRotationEmployeeId(rotationEmployee.getId());
+        shiftRestService.updateShift(TENANT_ID, testUpdateShift);
+
+        ShiftView retrievedShift = shiftRestService.getShift(TENANT_ID, testUpdateShift.getId());
+        assertClientResponseOk();
+        assertThat(retrievedShift).isNotNull().isEqualToComparingOnlyGivenFields(testUpdateShift, shiftViewNonIndictmentFields);
+
+        boolean result = shiftRestService.removeShift(TENANT_ID, retrievedShift.getId());
+        assertThat(result).isTrue();
+        assertClientResponseOk();
+
+        shifts = shiftRestService.getShifts(TENANT_ID);
+        assertThat(shifts).isEmpty();
+    }
+
+    @Test
+    public void testShiftWithDaylightSavingTime() {
+        createTestTenant(ZoneId.of("America/New_York"));
+        Spot spot = createSpot("spot");
+        Employee employee = createEmployee("employee");
+        LocalDateTime startDateTime = LocalDateTime.of(2018, 3, 10, 23, 0, 0, 0);
+        LocalDateTime endDateTime = startDateTime.plusHours(8);
+
+        ShiftView testAddShift = new ShiftView(TENANT_ID, spot, startDateTime, endDateTime);
+        testAddShift.setEmployeeId(employee.getId());
+
+        ShiftView addedShift = shiftRestService.addShift(TENANT_ID, testAddShift);
+        ShiftView retrievedShift = shiftRestService.getShift(TENANT_ID, addedShift.getId());
+        assertClientResponseOk();
+        assertThat(retrievedShift).isNotNull().isEqualToComparingOnlyGivenFields(testAddShift, shiftViewNonIndictmentFields);
+
+        // Trigger an indictment so we can read zone offset
+        LocalDateTime conflictingStartDateTime = endDateTime.plusHours(9);
+        LocalDateTime conflictingEndDateTime = conflictingStartDateTime.plusHours(8);
+
+        ShiftView testConflictingShift = new ShiftView(TENANT_ID, spot, conflictingStartDateTime, conflictingEndDateTime);
+        testConflictingShift.setEmployeeId(employee.getId());
+        addedShift = shiftRestService.addShift(TENANT_ID, testConflictingShift);
+        ShiftView retrievedConflictingShift = shiftRestService.getShift(TENANT_ID, addedShift.getId());
+
+        assertThat(retrievedShift.getEndDateTime().plusHours(9)).isEqualTo(retrievedConflictingShift.getStartDateTime());
+        assertThat(retrievedConflictingShift.getShiftEmployeeConflictList()).hasSize(1);
```

Comment:
Running the test locally, this assertions fails on size being 3. Could be connected to time zones? Replacing `createTestTenant` in the beginning of the test with Europe/Paris zone got me over this assertion (but failed on line 184)

Reply:
In order to test time-zones, I added two shifts with the same employee:
Shift 1: from 2018-03-10T23:00 to 2018-03-11T07:00 Local Time
Shift 2: from 2018-03-11T16:00 to 2018-03-11T24:00 Local Time

In the America/New_York time-zone, on 2018-03-11, at 2am, clocks were set forward 1 hour.
Thus, the actual shifts, in EST (the one that constant), would be:
Shift 1: from 2018-03-10T23:00 to 2018-03-11T08:00 EST
Shift 2: from 2018-03-11T17:00 to 2018-03-12T01:00 EST

This should cause only the "No 2 shifts within 10 hours from each other" rule to trigger, since Shift 2 starts 9 hours after Shift 1 ends. The "At most one shift assignment per day per employee" should not be triggered, since Shift 1 starts on 2018-03-10, while Shift 2 starts on 2018-03-11. Only the "No 2 shifts within 10 hours from each other" and "At most one shift assignment per day per employee" constraint matches are in `ShiftEmployeeConflictList`. The rule "No 2 shifts within 10 hours from each other", if matched, will have 1 constraint match (since it checks if the right shift ends before the left shift ends), and the "At most one shift assignment per day per employee", if matched, will have 2 constant matches (since it does not check if the right shift ends before the left shift ends).

Thus, the only way `size` could be 3 over there is if Shift 1 started on the same day as Shift 2 on the server, which should not be the case.

So either:
1: Your "America/New_York" time-zone is not up-to-date
or 2: The database is persisting date-times incorrectly.

This test passes for me, but I am situated in "America/Toronto" (which means my database will also be "America/Toronto") (which, in turn, has the same time-zone transitions as "America/New_York")

Reply:
Can confirm database is not persisting date-times correctly. Set time-zone to "Asia/Tokyo" and got a different assertion failed:
```
Expecting:
 <+09:00>
not to be equal to:
<+09:00>
```
Probably should do the same test in two different VERY DIFFERENT time-zones to test this issue.

Reply:
We have two choices here:
1. Update `Shift` to be in the correct time-zone in indictment
or 2. Change `Shift` to `ShiftView` and move test to server (note: ShiftView is in local time and loses information as a result)/ @rsynek, @ge0ffrey, @yurloc WDYT?